Source/DafnyRuntime/DafnyRuntime.py

```diff
@@ -374,11 +374,13 @@ def IntegerRange(lo, hi):
 class Doubler:
     def __init__(self, start):
         self.start = start
+        self.i = start
     def __iter__(self):
-        return self
+        return Doubler(self.start)
```

Comment:
I think that's not right either, because now `iter(iter(x))` doesn't behave the same as `iter(x)` (IOW, your object behaves as a mix between an iterator and an iterable).  Does the following work?

```
class Doubler:
    def __init__(self, start):
        self.start = start

    def __iter__(self):
        start = self.start
        while True:
            yield start
            start *= 2
```

(The one thing it's missing is the `__call__ == __next__`, but I'm not sure why we want to have that)

Reference here: https://stackoverflow.com/questions/45685538/whats-the-advantage-of-using-yield-in-iter